File: DeepSpaceRuby/lib/Damage.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
module Deepspace
  require_relative 'DamageToUI'
 
  
  class Damage
    
    attr_reader:nShields
    
    
    
    
   
    def copy()
      "No implementado"
    end
    
    def getUIversion
      "Abstracto"
    end
    
 
    def discardWeapon(w)
      "Abstracto"
    end
    
    def discardShieldBooster
      if @nShields >0
        @nShields=@nShields-1
      end
    end
    
    def hasNoEffect
      "Abstracto"
    end
    
    def to_s
      "NWeapons:#{@nWeapons} \n NShields:#{@nShields} \n Weapons:#{@weapons}"
    end
    
    def adjust(w,s)
       "Abstracto"
    end

    protected
    def initialize(s)
      @nShields=s
    end
   
  end
end

File: DeepSpaceJava/src/deepspace/Damage.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package deepspace;
import java.util.ArrayList;
import java.util.Collections;
        

/**
 *
 * @author laura
 */
public abstract class  Damage {
    protected int nShields;
    protected int nWeapons;
    protected ArrayList<WeaponType> weapons;
    
    Damage(int w, int s){
        nWeapons=w;
        nShields=s;
        weapons=null;
    }
    
    Damage(ArrayList<WeaponType> wl, int s){
        nWeapons=-1;
        nShields=s;
        weapons=wl;
    }
    
    public abstract Damage copy();
    
    
    public abstract Damage adjust(ArrayList<Weapon> w, ArrayList<ShieldBooster> s);
       

    
    

    
    void discardShieldBooster(){
     if(nShields >0)
        nShields-=1;
    }
    
     abstract void discardWeapon(Weapon w);
    
  
    abstract DamageToUI getUIversion();
    
    public int getNShields() {
        return nShields;
    }




      
    public abstract boolean hasNoEffect();
    
    
    @Override
    public String toString() {
        return "Damage{" + "nShields=" + nShields + ", nWeapons=" + nWeapons + ", weapons=" + weapons + '}';
    }
    
    
}
